src/t8_forest/t8_forest_partition.h

```diff
@@ -36,11 +36,14 @@ T8_EXTERN_C_BEGIN ();
 /**
  * Populate a forest with the partitioned elements of forest->set_from.
  * Currently the elements are distributed evenly (each element has the same weight).
- * 
+ *
  * \param [in,out]  forest  The forest.
+ * \param [in]      weight_callback A callback function defining element weights for the partitioning
+ * \pre \a weight_callback must be free of side-effects, the behavior is undefined otherwise
+ * \note If \a weight_callback is null, then all the elements are assumed to have the same weight
 */
 void
-t8_forest_partition (t8_forest_t forest);
+t8_forest_partition (t8_forest_t forest, t8_weight_fcn_t *weight_callback);
```

Comment:
Does it make sense to add the `weight_callback` as an extra argument here? To me, it seems like an unnecessary duplication, given that the forest already has the new member `weight_function`. 
My suggestion would be to omit the extra argument here and then use `forest->weight_function` within `t8_forest_partition` rather than the function argument? Otherwise my impression is that we are bound to sooner or later run into hard-to-debug problems of the argument and the member being out of sync.

What do you think?

```suggestion
t8_forest_partition (t8_forest_t forest);
```